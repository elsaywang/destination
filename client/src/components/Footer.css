:local(.footer) {
    bottom: 0;
    width: 100%;
    left: 0;
    right: 0;
    position: absolute;
    box-sizing: border-box;
    border-top: 0.0625rem solid var(--spectrum-global-color-grey300);
    padding: 0.9375rem;
    color: var(--spectrum-global-color-grey500);
    overflow: hidden;
}
:local(.footerLinks) {
    float: left;
    display: inline;
    line-height: 1.5rem;
    color: var(--spectrum-global-color-grey500);
}
:local(.footerLinkItem) {
    padding-right: 1.25rem;
    text-decoration: none;
    vertical-align: middle;
}
:local(.footerLabel) {
    display: inline;
}
:local(.footerLanguage) {
    display: inline;
    color: var(--spectrum-semantic-informative-color-text-small);
}
:local(.footerCopyRight) {
    float: right;
    line-height: 1.5rem;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}
