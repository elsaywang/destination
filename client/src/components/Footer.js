import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { FormItem } from '@react/react-spectrum/Form';
import { getFooterCopyRightMessage, languageOption } from '../constants/footerOptions';
import { footerUrls } from '../utils/urls';
import styles from './Footer.css';
import classNames from 'classnames';

function Footer(props) {
    return (
        <footer className={styles.footer}>
            <div className={styles.footerLinks}>
                {footerUrls.map(({ item, url }) => (
                    <a
                        key={item}
                        target="_blank"
                        className={classNames(styles.footerLinks, styles.footerLinkItem)}
                        href={url}>
                        {item}
                    </a>
                ))}
                <label htmlFor={styles.footerLanguage} className={styles.footerLabel}>
                    {languageOption.label}
                </label>
                <a href="#" className={styles.footerLanguage}>
                    {languageOption.option}
                </a>
            </div>
            <div className={styles.footerCopyRight}>{getFooterCopyRightMessage()}</div>
        </footer>
    );
}

export default Footer;
