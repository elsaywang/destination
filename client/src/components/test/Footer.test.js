import React from 'react';
import { shallow } from 'enzyme';
import Footer from '../Footer';

describe('<Footer/> component', () => {
    const wrapper = shallow(<Footer />);
    describe('rendering', () => {
        it('matches snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });

        it('should have 4 footer items link', () => {
            expect(wrapper.find('a').length).toEqual(4);
        });

        it('should have 1 lable for `footerLanguage`', () => {
            expect(wrapper.find('label').length).toEqual(1);
            expect(wrapper.find('[htmlFor="footerLanguage"]').length).toEqual(1);
        });

        it('should have 1 footerCopyRight element', () => {
            expect(wrapper.find('div.footerCopyRight').length).toEqual(1);
        });
    });
});
