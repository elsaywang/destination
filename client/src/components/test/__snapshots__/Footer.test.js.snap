// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> component rendering matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Footer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="footerLinks"
>
          <a
                    className="footerLinks footerLinkItem"
                    href="https://marketing.adobe.com/resources/help/en_US/aam/"
                    target="_blank"
          >
                    Help
          </a>
          <a
                    className="footerLinks footerLinkItem"
                    href="http://www.adobe.com/go/marketingcloud_terms_of_use"
                    target="_blank"
          >
                    Terms of Use
          </a>
          <a
                    className="footerLinks footerLinkItem"
                    href="http://www.adobe.com/go/marketingcloud_privacy"
                    target="_blank"
          >
                    Privacy Policy
          </a>
          <label
                    className="footerLabel"
                    htmlFor="footerLanguage"
          >
                    Language: 
          </label>
          <a
                    className="footerLanguage"
                    href="#"
          >
                    English
          </a>
</div>,
        <div
          className="footerCopyRight"
>
          © 2018 Adobe Systems Incorporated. All Rights Reserved.
</div>,
      ],
      "className": "footer",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <a
                className="footerLinks footerLinkItem"
                href="https://marketing.adobe.com/resources/help/en_US/aam/"
                target="_blank"
>
                Help
</a>,
              <a
                className="footerLinks footerLinkItem"
                href="http://www.adobe.com/go/marketingcloud_terms_of_use"
                target="_blank"
>
                Terms of Use
</a>,
              <a
                className="footerLinks footerLinkItem"
                href="http://www.adobe.com/go/marketingcloud_privacy"
                target="_blank"
>
                Privacy Policy
</a>,
            ],
            <label
              className="footerLabel"
              htmlFor="footerLanguage"
>
              Language: 
</label>,
            <a
              className="footerLanguage"
              href="#"
>
              English
</a>,
          ],
          "className": "footerLinks",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Help",
            "nodeType": "host",
            "props": Object {
              "children": "Help",
              "className": "footerLinks footerLinkItem",
              "href": "https://marketing.adobe.com/resources/help/en_US/aam/",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Help",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": "Terms of Use",
            "nodeType": "host",
            "props": Object {
              "children": "Terms of Use",
              "className": "footerLinks footerLinkItem",
              "href": "http://www.adobe.com/go/marketingcloud_terms_of_use",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Terms of Use",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": "Privacy Policy",
            "nodeType": "host",
            "props": Object {
              "children": "Privacy Policy",
              "className": "footerLinks footerLinkItem",
              "href": "http://www.adobe.com/go/marketingcloud_privacy",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Privacy Policy",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Language: ",
              "className": "footerLabel",
              "htmlFor": "footerLanguage",
            },
            "ref": null,
            "rendered": "Language: ",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "English",
              "className": "footerLanguage",
              "href": "#",
            },
            "ref": null,
            "rendered": "English",
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "© 2018 Adobe Systems Incorporated. All Rights Reserved.",
          "className": "footerCopyRight",
        },
        "ref": null,
        "rendered": "© 2018 Adobe Systems Incorporated. All Rights Reserved.",
        "type": "div",
      },
    ],
    "type": "footer",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="footerLinks"
>
            <a
                        className="footerLinks footerLinkItem"
                        href="https://marketing.adobe.com/resources/help/en_US/aam/"
                        target="_blank"
            >
                        Help
            </a>
            <a
                        className="footerLinks footerLinkItem"
                        href="http://www.adobe.com/go/marketingcloud_terms_of_use"
                        target="_blank"
            >
                        Terms of Use
            </a>
            <a
                        className="footerLinks footerLinkItem"
                        href="http://www.adobe.com/go/marketingcloud_privacy"
                        target="_blank"
            >
                        Privacy Policy
            </a>
            <label
                        className="footerLabel"
                        htmlFor="footerLanguage"
            >
                        Language: 
            </label>
            <a
                        className="footerLanguage"
                        href="#"
            >
                        English
            </a>
</div>,
          <div
            className="footerCopyRight"
>
            © 2018 Adobe Systems Incorporated. All Rights Reserved.
</div>,
        ],
        "className": "footer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <a
                  className="footerLinks footerLinkItem"
                  href="https://marketing.adobe.com/resources/help/en_US/aam/"
                  target="_blank"
>
                  Help
</a>,
                <a
                  className="footerLinks footerLinkItem"
                  href="http://www.adobe.com/go/marketingcloud_terms_of_use"
                  target="_blank"
>
                  Terms of Use
</a>,
                <a
                  className="footerLinks footerLinkItem"
                  href="http://www.adobe.com/go/marketingcloud_privacy"
                  target="_blank"
>
                  Privacy Policy
</a>,
              ],
              <label
                className="footerLabel"
                htmlFor="footerLanguage"
>
                Language: 
</label>,
              <a
                className="footerLanguage"
                href="#"
>
                English
</a>,
            ],
            "className": "footerLinks",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Help",
              "nodeType": "host",
              "props": Object {
                "children": "Help",
                "className": "footerLinks footerLinkItem",
                "href": "https://marketing.adobe.com/resources/help/en_US/aam/",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Help",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": "Terms of Use",
              "nodeType": "host",
              "props": Object {
                "children": "Terms of Use",
                "className": "footerLinks footerLinkItem",
                "href": "http://www.adobe.com/go/marketingcloud_terms_of_use",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Terms of Use",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": "Privacy Policy",
              "nodeType": "host",
              "props": Object {
                "children": "Privacy Policy",
                "className": "footerLinks footerLinkItem",
                "href": "http://www.adobe.com/go/marketingcloud_privacy",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Privacy Policy",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Language: ",
                "className": "footerLabel",
                "htmlFor": "footerLanguage",
              },
              "ref": null,
              "rendered": "Language: ",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "English",
                "className": "footerLanguage",
                "href": "#",
              },
              "ref": null,
              "rendered": "English",
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "© 2018 Adobe Systems Incorporated. All Rights Reserved.",
            "className": "footerCopyRight",
          },
          "ref": null,
          "rendered": "© 2018 Adobe Systems Incorporated. All Rights Reserved.",
          "type": "div",
        },
      ],
      "type": "footer",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
